components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +248,190 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
+        when(requestMsgCtx.getAuthorizedUser()).thenReturn(user);
+        when(user.isFederatedUser()).thenReturn(false);
+
+        OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO = mock(OAuth2AccessTokenReqDTO.class);
+        when(requestMsgCtx.getOauth2AccessTokenReqDTO()).thenReturn(oAuth2AccessTokenReqDTO);
+        when(oAuth2AccessTokenReqDTO.getTenantDomain()).thenReturn(SAMPLE_TENANT_DOMAIN);
+        when(oAuth2AccessTokenReqDTO.getClientId()).thenReturn(SAMPLE_CLIENT_ID);
+
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        applicationManagementService = mock(ApplicationManagementService.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+
+        getMockResourceFromScopes();
+
+        samlAssertionClaimsCallback.handleCustomClaims(jwtClaimsSet, requestMsgCtx);
+        assertEquals(jwtClaimsSet.getAllClaims().size(), 8, "Claims are not successfully set.");
+    }
+
+    @Test
+    public void testHandleClaimsForOAuthAuthzReqMessageContextNullAccessToken() throws Exception {
+        jwtClaimsSet = new JWTClaimsSet();
+        oAuthAuthzReqMessageContext = mock(OAuthAuthzReqMessageContext.class);
+        String[] approvedScopes = {"openid", "testScope1", "testScope2"};
+        when(oAuthAuthzReqMessageContext.getApprovedScope()).thenReturn(approvedScopes);
+        when(oAuthAuthzReqMessageContext.getProperty(OAuthConstants.ACCESS_TOKEN))
+                .thenReturn(null);
+
+        OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO = new OAuth2AuthorizeReqDTO();
+        when(oAuthAuthzReqMessageContext.getAuthorizationReqDTO()).thenReturn(oAuth2AuthorizeReqDTO);
+
+        AuthenticatedUser authenticatedUser = mock(AuthenticatedUser.class);
+        oAuth2AuthorizeReqDTO.setUser(authenticatedUser);
+        when(authenticatedUser.isFederatedUser()).thenReturn(false);
+
+        when(authenticatedUser.getUserName()).thenReturn("user1234");
+        when(authenticatedUser.getUserStoreDomain()).thenReturn("abc.com");
+
+        ApplicationManagementService mockApplicationManagementService = mock(ApplicationManagementService.class);
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(mockApplicationManagementService);
+
+        when(mockApplicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+
+        ServiceProvider serviceProvider = new ServiceProvider();
+        ClaimConfig claimConfig = new ClaimConfig();
+        ClaimMapping claimMap1 = ClaimMapping.build("http://www.wso2.org/claims/email", "http://www.abc.com/claims/email", "sample@abc.com", false);
```

Comment:
In UX point of view, its better to use some valid emails which we are using on documentation samples

Reply:
I initially used "john@abc.com"/"john@wso2.com", then after discussing with IAM team, we decided not to put valid email addresses but dummy ones because there might be users for that email address. The dummy email should satisfy the regex that an email should have. Then came up with "sample@abc.com".